ethcore/src/migrations/account_bloom.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Bloom upgrade
+
+use client::{DB_COL_EXTRA, DB_COL_HEADERS, DB_NO_OF_COLUMNS, DB_COL_STATE};
+use state_db::{ACCOUNT_BLOOM_SPACE, DEFAULT_ACCOUNT_PRESET, StateDB};
+use util::trie::TrieDB;
+use views::HeaderView;
+use bloomfilter::Bloom;
+use util::migration::Error;
+use util::journaldb;
+use util::{H256, FixedHash, BytesConvertable};
+use util::{Database, DatabaseConfig, DBTransaction, CompactionProfile};
+use std::path::Path;
+
+/// Account bloom upgrade routine. If bloom already present, does nothing.
+/// If database empty (no best block), does nothing.
+/// Can be called on upgraded database with no issues (will do nothing).
+pub fn upgrade_account_bloom(db_path: &Path) -> Result<(), Error> {
+	let path = try!(db_path.to_str().ok_or(Error::MigrationImpossible));
+	trace!(target: "migration", "Account bloom upgrade at {:?}", db_path);
+
+	let source = try!(Database::open(&DatabaseConfig {
+		max_open_files: 64,
+		cache_size: None,
+		compaction: CompactionProfile::default(),
+		columns: DB_NO_OF_COLUMNS,
+		wal: true,
+	}, path));
+
+	let best_block_hash = match try!(source.get(DB_COL_EXTRA, b"best")) {
+		// no migration needed
+		None => {
+			trace!(target: "migration", "No best block hash, skipping");
+			return Ok(());
+		},
+		Some(hash) => hash,
+	};
+	let best_block_header = match try!(source.get(DB_COL_HEADERS, &best_block_hash)) {
+		// no best block, nothing to do
+		None => {
+			trace!(target: "migration", "No best block header, skipping");
+			return Ok(())
+		},
+		Some(x) => x,
+	};
+	let state_root = HeaderView::new(&best_block_header).state_root();
+
+	if StateDB::check_bloom_exists(&source) {
+		// bloom already exists, nothing to do
+		trace!(target: "migration", "Bloom already present, skipping");
+		return Ok(())
+	}
+
+	println!("Adding accounts bloom (one-time upgrade). Please don't close parity.");
```

Comment:
use info! instead of println!
"Please don't close" should be irrelevant. The whole upgrade should be one transaction or pa process that can be canceled